pilot/pkg/xds/ads.go

```diff
@@ -461,6 +468,14 @@ func (s *DiscoveryServer) initConnection(node *core.Node, con *Connection) error
 	con.ConID = connectionID(node.Id)
 	con.node = node
 
+	if features.EnableXDSIdentityCheck && con.Identities != nil {
```

Comment:
shouldn't this reject unauthenticated calls if I'm setting the feature to be enabled?

Reply:
My concern here is that
1. We need this check for two features coming along (Istiod SDS and VM auto registration). Therefor we want this to always exist
2. Users may be using plaintext port and I am worried about breaking them

We may want a second flag that disables plaintext though to lock things down?

Reply:
We need to be careful here. It's too implicit that "identities == nil" == "authentication skipped", though it might be true. Can we add a check on the return not being nil here:
https://github.com/istio/istio/blob/master/pilot/pkg/xds/authentication.go#L56